safehttp/mux.go

```diff
@@ -165,7 +165,8 @@ func (m methodHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	for _, intercep := range m.muxInterceps {
-		if res := intercep.Before(rw, ir); res.written {
+		intercep.Before(rw, ir)
+		if rw.written {
 			return
```

Comment:
Hm... What if other interceptors don't write? Should they run?

Also, what if the handler doesn't write?

Reply:
A handler that doesn't write any response seems rather pointless. Maybe we should panic if nothing has been written when the handler is done executing?

If we assume that `Before` will only write something when there is some sort of error, then I don't see the point of running more `Before` methods when we already know the response is an error.

I can't come up with any `Before`-phases that might want to write something that isn't an error. But if they exist then we might want to continue running other `Before`-phases. If in that case a later `Before` method also writes a response then it's OK if it panics since panics will be recovered and a 500 Internal Server Error will be returned.

So should I remove the
```go
if rw.written {
    return
}
```
?